app/controllers/merchant/discounts_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Merchant::DiscountsController < Merchant::BaseController
+  def index
+    @discounts = Discount.all
+  end
+
+  def new
+    merchant = Merchant.find(current_user[:merchant_id])
+    @discount = merchant.discounts.new
```

Comment:
This looked weird at first but it needs to be an instance var because it is passed in through a form_for